services/ui_backend_service/data/cache/store.py

```diff
@@ -126,65 +100,21 @@ async def get_run_parameters(self, flow_name, run_number):
 
         Returns
         -------
-        [
-            {
-                "name": "name_of_parameter",
-                "value": "value-of-parameter-as-string"
+        {
+            "name_of_parameter": {
+                "value": "value_of_parameter"
             }
-        ]
+        }
         OR None
         '''
-        run_id_key, run_id_value = translate_run_key(run_number)
-
-        # '_parameters' step has all the parameters as artifacts. only pick the
-        # public parameters (no underscore prefix)
-        db_response, *_ = await self._artifact_table.find_records(
-            conditions=[
-                "flow_id = %s",
-                "{run_id_key} = %s".format(run_id_key=run_id_key),
-                "step_name = %s",
-                "name NOT LIKE %s",
-                "name <> %s",
-                "name <> %s"
-            ],
-            values=[
-                flow_name,
-                run_id_value,
-                "_parameters",
-                r"\_%",
-                "name",  # exclude the 'name' parameter as this always exists, and contains the FlowName
-                "script_name"  # exclude the internally used 'script_name' parameter.
-            ],
-            fetch_single=False,
-            expanded=True
-        )
+        db_response, *_ = await self._artifact_table.get_run_parameter_artifacts(flow_name, run_number)
```

Comment:
this basically uses find_records under the hood, which accepts a postprocessing callable as a parameter. could be further simplified by passing a refiner in I guess.. but might convolute the issue further. Much cleaner this way than previously anyway :)

Reply:
Very valid comment